engine/util/src/main/scala/pl/touk/nussknacker/engine/util/Implicits.scala

```diff
@@ -22,4 +23,11 @@ object Implicits {
       else Some(s)
     }
   }
+
+  implicit class RichFuture[A](future: Future[A]) {
+    def effect(f: () => Unit)(implicit ec: ExecutionContext): Future[A] = for {
```

Comment:
Maybe `withSideEffect` or at least `withEffect` - to be visible that it doesn't only add effect but also returns Future

Reply:
`withSideEffect`sounds great